src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,852 @@
+```
+
+5. Select the top 15 taxa and save them to the `css_tse` object. We keep the `glom_tse` and `css_tse` objects separate as it will make things easier later.
+
+```{r}
+# Calculate the taxa sums by row
+taxa_sums <- rowSums(assay(glom_tse, "css"))
+
+# Select the top 15 taxa
+top15_taxa <- names(sort(taxa_sums, decreasing = TRUE)[1:15])
+
+# Create the new TreeSummarizedExperiment object with assays, colData, and rowData
+css_tse <- TreeSummarizedExperiment(
+  assays = list(css = assay(glom_tse, "css")[top15_taxa, ]),
+  colData = colData(glom_tse),
+  rowData = rowData(glom_tse)[top15_taxa, ]
+)
+```
+
+## Computing alpha diversity indices
+Alpha diversity is a measure of species diversity in a particular area or an ecosystem. In terms of microbial ecology, analyzing the sample diversity of sequencing data is commonly the first step in assessing differences between microbial environments. Some of the most popular alpha diversity indices reported in microbial community analyses are Chao, abundance-based coverage estimators (ACE), Simpson, and Shannon-Weaver.
+
+Chao1 and ACE are nonparametric estimators that describe diversity as the max number of expected species in the sample (species richness). They consider the proportion of species that have been observed before (“recaptured”) to those that are observed only once. Both Chao1 and ACE underestimate true richness at low sample sizes.
+
+On the other hand, Simpson and Shannon-Weaver, consider relative abundances and depends on both, species richness and the evenness (or equitableness), with which individuals are distributed among the different species. However, both metrics have specific biases. The Shannon-Weaver index places a greater weight on species richness, whereas the Simpson index considers evenness more than richness in its measurement.
+
+More discussion and examples illustrating how sample size and normalization methods can affect alpha diversity metrics in references 4 and 5.
+
+In this example, we're going to calculate the diversity measures for our data with no normalization, our CSS normalized data, and our rarefied data.
+
+1. First, let's define a function for calculating the diversity measures for a given assay.
+
+```{r}
+calculate_diversity_measures <- function(tse, assay_type) {
+  colData(tse)[[paste0("observed_", assay_type)]] <- estimateRichness(tse, index = "observed", assay.type = assay_type)$observed
+  colData(tse)[[paste0("chao1_", assay_type)]] <- estimateRichness(tse, index = "chao1", assay.type = assay_type)$chao1
+  colData(tse)[[paste0("ace_", assay_type)]] <- estimateRichness(tse, index = "ace", assay.type = assay_type)$ace
+  colData(tse)[[paste0("shannon_", assay_type)]] <- estimateDiversity(tse, index = "shannon", assay.type = assay_type)$shannon
+  colData(tse)[[paste0("simpson_", assay_type)]] <- estimateDiversity(tse, index = "inverse_simpson", assay.type = assay_type)$inverse_simpson
+  
+  return(tse)
+}
```

Comment:
This should work with mia::addAlpha()

Reply:
perfect yeah I though there must be a simpler method but couldn't see it in OMA

Reply:
Has this now been changed? I do not see the changes yet here.